### DISCUSSION TITLE
Can I configure to group PRs only starting with a specific version ?

### DISCUSSION THREAD

**Discussion by onigoetz** [renovatebot/renovate#20377] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

Hi,

I'm trying to understand if the `packageRules` `groupName` feature could work conditionally based on some critiera.

A bit of context:
in the Java world there is an ongoing migration from Javax to Jakarta, in short it means that a lot of dependencies have to be changed at the same time to a new major version.
Usually, these libraries shouldn't be grouped in a single PR, but for this specific migration we would like to bundle all the updates in a single PR (anyway, if for example two dependencies need to be updated, doing two PRs for them will fail them in CI ...)

I know about the `allowedVersions` rule, but it's not what I'm looking for.

Example 

```json
{
  "packageRules": [
    {
      "groupName": "jakarta",
      "packagePatterns": ["^org.springframework"],
      "applyToVersion": ">6.0.0"
    },
    {
      "groupName": "jakarta",
      "packagePatterns": ["^org.hibernate:hibernate-validator"],
      "applyToVersion": ">7.0.0"
    },
}
```

This would make so that any dependency of `org.springframework...` with a version higher than 6.0.0 to have a groupName of `jakarta` but the rule wouldn't apply if the version is below 6.0.0

Is there a way to achieve this ?

**Comment by rarkins** [comment]
If you match for example <6 current version and updateType=major, will that achieve the same thing? You could match current version 5.x if you want to be sure

**Reply by onigoetz** [reply]
I'm not sure I understand your answer.

`updateType=major` might achieve the same, I should try that.

> You could match current version 5.x if you want to be sure
How would I do that? I don't want to limit with `allowedVersions` as all other versions are still valid

**Reply by viceice** [reply]
you can also use `matchCurrentVersion` or `matchCurrentValue` , check the docs. 😉

**Reply by onigoetz** [reply]
Oh indeed, I didn't see this before, thanks for the replies :D

**Reply by onigoetz** [reply]
Hi,
I'm looking into this again and have more questions on the usage of `matchCurrentVersion` 

My use case in the initial question is that I would like to apply the `packageRule` on the updated version, but `matchCurrentVersion` applies only to the, well ..., current version.

Is there a way to create a rule along the lines of: "If you're currently using Spring 5, the PR to upgrade to Spring 6 should be grouped in 'jakarta', but should only be triggered by the Dependency Dashboard"

```
{
  "packageRules": [
    {
      "groupName": "jakarta",
      "packagePatterns": ["^org.springframework"],
      "matchUpdateVersion": ">=6.0.0",
      "enabled": false
    }
}
```

Because I would like all minor/patch updates of that dependency to be grouped into "all non major". And the updates for Jakarta have to be done all at the same time otherwise they fail

But I have the feeling that the `matchUpdateVersion` brick is the missing link I need for this. Or is there another way to do this ?

**Reply by rarkins** [reply]
Is there anything preventing you from using `matchUpdateTypes` as part of your logic? It does not match the exact updated version but it might product the same result if there is no v7 in your example

**Reply by onigoetz** [reply]
In 80% of the cases `matchUpdateTypes` would work.
But in some cases, the actual major change is done in a minor version. For example, Logback has done the update from Javax to Jakarta in version 1.4: https://github.com/qos-ch/logback#java-ee-and-jakarta-ee-versions

In this case a `"matchUpdateVersion": ">=1.4.0"` would help
